db/migrate/20211126135622_add_targets_count_to_users.rb

```diff
@@ -0,0 +1,5 @@
+class AddTargetsCountToUsers < ActiveRecord::Migration[6.1]
+  def change
+    add_column :users, :targets_count, :integer
```

Comment:
I think we don't need this. keeping a count attribute up-to-date adds unnecessary complexity to your code.
Instead of doing this, you can add a `has_many :targets` association in the`User` model. Then you can simply do `user.targets.count`

Reply:
Since we're checking how many targets does a user have, each time a new target is created, don't you think doing a COUNT() at the db level could bring some performance issues?

Reply:
Not really, a rails count should be O(1) constant time. Doesn't impact the database performance. Unless you create thousands of targets per second.

Reply:
We are also using `counter_cache` which should also help with any peformance impact, so I agree we should not be having this attribute due to the complexity and error prone nature of doing it this way

Reply:
Oh, this migration is needed for the counter_cache, the purpose of this migration is to implement the `counter_cache`

Reply:
Calling `.count` upon a collection of records, it passes the call to the database, then it performs a `COUNT()` function, which AFAIK is O(n). I could be wrong tho.

That why I though implementing the counter_cache was needed in this case.

Reply:
Oh, you are right, it is indeed needed for counter_cache, good call!!

Reply:
oh, I didn't notice the intention to use counter_cache here, that makes sense to me, good job! 👏